### PR TITLE
Add testsets to Givens tests

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: test/linalg/givens.jl [hunk 1]
```diff
@@ -2,19 +2,14 @@
...
-    debug && println("elty is $elty")
-
+@testset for elty in (Float32, Float64, Complex64, Complex128)
     if elty <: Real
         A = convert(Matrix{elty}, randn(10,10))
     else
         A = convert(Matrix{elty}, complex.(randn(10,10),randn(10,10)))
     end
-    for Atype in ("Array", "SubArray")
+    @testset for Atype in ("Array", "SubArray")
```
Comment (fredrikekre) [inline test/linalg/givens.jl]: Perhaps
```julia
@testset for A in (A, view(A, 1:10, 1:10)
...
```
and remove the `if-else-end` below?

File: test/linalg/givens.jl [hunk 2]
```diff
@@ -51,25 +47,25 @@ for elty in (Float32, Float64, Complex64, Complex128)
...
-            x = A[:, 1]
-        else
-            x = view(A, 1:10, 1)
-        end
-        G, r = givens(x[2], x[4], 2, 4)
-        @test (G*x)[2] ≈ r
-        @test abs((G*x)[4]) < eps(real(elty))
-        @inferred givens(x[2], x[4], 2, 4)
+        @testset "Givens * vectors" begin
+            if Atype == "Array"
```
Comment (fredrikekre) [inline test/linalg/givens.jl]: Need to update here also, if you go with the comment above.

**Review by fredrikekre — APPROVED** [review]
just some small suggestions

**Comment by kshyatt** [comment]
@fredrikekre thanks for pointing that out - look good?

**Comment by kshyatt** [comment]
We had a build timeout and an `arnoldi` fail on Travis so I restarted the build to see if I can get at least one to pass.

**Comment by kshyatt** [comment]
*sigh* apparently not. Do we want to just wait on these?

**Comment by fredrikekre** [comment]
Relevant tests have passed several times on all buildbots

**Pull request merged**